### PR TITLE
Ensure framework ID is set once.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
Relates to https://github.com/jenkinsci/configuration-as-code-plugin/issues/1392.

JIRA issues: D2IQ-68537

Co-Authored-By: Joseph Petersen <josephp90@gmail.com>

### CODE REVIEW

File: src/test/java/org/jenkinsci/plugins/mesos/MesosCloudTest.java [hunk 1]
```diff
@@ -82,4 +83,14 @@ void serializationRoundTrip(TestUtils.JenkinsRule j)
...
   }
+
+  @Test
+  void configureAsCode(TestUtils.JenkinsRule j) throws IOException {
+    final String config =
+        IOUtils.resourceToURL("configuration.yaml", this.getClass().getClassLoader())
+            .toExternalForm();
+    ConfigurationAsCode.get().configure(config);
+
+    assertThat(j.jenkins.clouds.getAll(MesosCloud.class), hasSize(1));
```
Comment (jeschkies) [inline src/test/java/org/jenkinsci/plugins/mesos/MesosCloudTest.java:94]: This needs to be extended a little.

File: src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java [hunk 2]
```diff
@@ -185,19 +176,11 @@ private Object readResolve() throws IOException {
...
-              this.master,
-              this.jenkinsURL,
-              this.agentUser,
-              this.frameworkName,
-              this.frameworkId,
-              this.role,
-              this.sslCert,
-              this.dcosAuthorization);
-      logger.info("Initialized Mesos API object after deserialization.");
+      MesosApi.getInstance(this);
```
Comment (jetersen) [inline src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java:179]: This seems ill adviced, same reason why it was ill adviced to do this in the constructor.

File: src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java [hunk 3]
```diff
@@ -135,24 +131,19 @@ public MesosCloud(
 
     this.agentUser = agentUser;
     this.role = role;
-    this.mesosAgentSpecTemplates = mesosAgentSpecTemplates;
+    this.mesosAgentSpecTemplates =
+        mesosAgentSpecTemplates != null ? mesosAgentSpecTemplates : Collections.emptyList();
```
Comment (jetersen) [inline src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java:135]: Hudson.util has an static method for fixing empty collections
https://github.com/jenkinsci/jenkins/blob/598d78b8b570b6eefffbc75b7df6dc0d29220232/core/src/main/java/hudson/Util.java#L1065-L1119
```suggestion
    this.mesosAgentSpecTemplates = Util.fixNull(mesosAgentSpecTemplates);
```

File: Dockerfile [hunk 4]
```diff
@@ -6,7 +6,7 @@ RUN gradle jpi
 
 
 # Stage: Prod
-FROM jenkins/jenkins:2.176.1 as prod
+FROM jenkins/jenkins:2.176.4 as prod
```
Comment (kaiwalyajoshi) [inline Dockerfile:9]: Might as well use the latest:
```suggestion
FROM jenkins/jenkins:2.222.3 as prod
```

File: src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java (hunk 2 above)
Comment (jeschkies) [reply src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java:179]: Thanks for the review. This line is not required.

File: Dockerfile (hunk 4 above)
Comment (jeschkies) [reply Dockerfile:9]: Cool. Are we bumping to the latest?

**Pull request merged**